lib/puppet/pops/types/types.rb

```diff
@@ -840,11 +1021,26 @@ def generalize
       def hashed_elements
         @hashed ||= @elements.reduce({}) {|memo, e| memo[e.name] = e; memo }
       end
- 
+
       def hash
         @elements.hash
       end
 
+      def iterable?
+        true
+      end
+
+      def iterable_type
+        if self == DEFAULT
+          PIterableType.new(PTupleType.new([PAnyType::DEFAULT], PHashType::TUPLE_SIZE))
```

Comment:
Do not understand, does this create an Iterable[Iterable[Any, what?? ) Think a comment is needed here